src/components/ProjectStatusIcon/index.jsx

```diff
@@ -0,0 +1,33 @@
+import cls from 'classnames';
+import {
+  CheckCircleIcon,
+  MinusCircleIcon,
+} from '@heroicons/react/24/solid';
+import styles from './styles.module.css';
+import ArchiveIcon from './ArchiveIcon';
+
+/**
+ *
+ * @param {Object} props
+ * @param {('md' | 'lg')} [props.size=md] Icon size. Medium or large
+ * @param {('published' | 'archive')} [props.status] Published or archive.
+ * Otherwise dispalys `"MinuseCircleIcon"`
+ * @returns {JSX.Element}
+ */
+export default function ProjectStatusIcon({ status, size = 'md' }) {
+  const iconClass = cls(styles.icon, styles[size], styles[status]);
+
+  const renderIcon = () => {
+    if (status === 'published') {
+      return <CheckCircleIcon />;
+    }
+    if (status === 'archive') {
+      return (
+        <ArchiveIcon />
+      );
+    }
+    return <MinusCircleIcon />;
+  };
```

Comment:
So in keeping with our only one return node per component, this should proabably be a map you set up outside this component

```
const ICONS = {
  published: CheckCirlceIcon,
  archive: ArchiveIcon,
  ...
}
```

Then in the component you can

```
const IconComponent = ICONS[status] || DefaultIcon
```